pymbolic/mapper/__init__.py

```diff
@@ -496,9 +496,9 @@ def map_call_with_kwargs(self, expr, *args, **kwargs):
         parameters = tuple([
             self.rec(child, *args, **kwargs) for child in expr.parameters
             ])
-        kw_parameters = {
+        kw_parameters = immutabledict({
                 key: self.rec(val, *args, **kwargs)
-                for key, val in expr.kw_parameters.items()}
+                for key, val in expr.kw_parameters.items()})
```

Comment:
What if the values are not hashable here? (I think that's going to be an issue with `IntG`)

Reply:
Then `IntG` needs to provide methods (`__hash__` and `update_persistent_hash`) to paper over that.

Reply:
(or get its act together)

Reply:
I don't think it can currently provide it's own hash because `expr_dataclass` comes after and it overwrites the `__hash__` implementation. (Right?)

But getting its act together sounds like a nicer goal :grin:

Reply:
Yeah, `IntG` often has a `dsource_vec` in there.

Reply:
Well, our hands may be a bit tied on making numpy arrays immutable. I've just added a widget that allows turning off the automatic `__hash__`.